MIGRATING_TO_V8.0.0.MD

```diff
@@ -0,0 +1,214 @@
+# Migrating to V8.0.0
+
+## Double encoding RQL values 
+From version 8.0.0 RQL values are now [double encoded](https://docs.extrahorizon.com/extrahorizon/additional-resources/resource-query-language-rql#double-encoding-of-special-characters) by default when using the rql builder, to support the use of special characters in rql operations.
+
+In your existing application it is likely that you have encoded values provided to the rql builder using methods such as `encodeURIComponent()`. To support the automatic double encoding of special characters, where applicable, these instances of encoding should be removed.
+- Before 8.0.0: `rqlBuilder().eq('diagnosis', encodeURIComponent('Hypertension - STAGE 1'))`
+- After 8.0.0: `rqlBuilder().eq('diagnosis', 'Hypertension - STAGE 1')`
+
+Any use of an rql builder that does not need to support special characters may remain as it is.
+
+### Reverting double encoding for a single rql builder
+You may disable double encoding for an rql builder by setting the option `rqlBuilder({ doubleEncodeValues: false })`
+- Before 8.0.0: `rqlBuilder().eq('diagnosis', encodeURIComponent('Hypertension - STAGE 1'))`
+- After 8.0.0: `rqlBuilder({ doubleEncodeValues: false }).eq('diagnosis', encodeURIComponent('Hypertension - STAGE 1'))`
+
+### Reverting double encoding for all rql builders
+The automatic double encoding of values can be reverted to behavior before 8.0.0 using the snippet `rqlBuilder.doubleEncodeValues = false`. This should normally exist alongside the declaration of your Extra Horizon SDK instance at the root of your application.
+- You may then enable double encoding for an rql builder `rqlBuilder({ doubleEncodeValues: true }).eq('diagnosis', 'Hypertension - STAGE 1')`
+- You may manually encode values for an rql builder `rqlBuilder().eq('diagnosis', encodeURIComponent('Hypertension - STAGE 1'))`
+
+For more information regarding manual double encoding please refer to the [RQL documentation](https://docs.extrahorizon.com/extrahorizon/additional-resources/resource-query-language-rql#double-encoding-of-special-characters).
+
+## Custom data normalization
+
+From version 8.0.0, the SDK will no longer normalize custom keys in requests and responses. This means that all custom keys will be sent and received as they are provided.
+
+Before version 8.0.0 all keys were normalized to snake_case in requests and camelCase in responses. This is useful because some of our services still return snake_case keys, while others return camelCase keys. 
+We wanted to keep this behaviour for all keys defined by Extra Horizon services, but we also wanted to allow custom keys to be sent and received as they are provided.
+
+### Changes to request data
+If you are providing custom keys in camel case before v8.0.0, they will now be stored as camel case.
+While before they would be stored as snake case for some requests.
```

Comment:
If you have provided custom keys in camel case before v8.0.0, they will now be stored in camel case instead of the previous snake case for some requests

OR 

Prior to v8.0.0, when custom keys were supplied in snake case, they were internally stored in camel case. However, starting from v8.0.0, custom keys are now preserved in the case provided during input.